src/components/SupplementalFiles/SupplementalFiles.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { useManifestState } from '../../context/manifest-context';
+import { getRenderingFiles, getSupplementingFiles } from '@Services/iiif-parser';
+import { fileDownload } from '@Services/utility-helpers';
+import './SupplementalFiles.scss';
+
+const SupplementalFiles = ({ itemHeading = "Item files", sectionHeading = "Section files", showHeading = true }) => {
+  const { manifest } = useManifestState();
+
+  const [manifestSupplementalFiles, setManifestSupplementalFiles] = React.useState();
+  const [canvasSupplementalFiles, setCanvasSupplementalFiles] = React.useState();
+
+  React.useEffect(() => {
+    if (manifest) {
+      let renderings = getRenderingFiles(manifest);
+
+      let manifestFiles = renderings.manifest;
+      setManifestSupplementalFiles(manifestFiles);
+
+      let annotations = getSupplementingFiles(manifest);
+      let canvasFiles = renderings.canvas;
+      canvasFiles.map((canvas, index) => canvas.files = canvas.files.concat(annotations[index].files));
+      canvasFiles = canvasFiles.filter(canvasFiles => canvasFiles.files.length > 0 );
+      setCanvasSupplementalFiles(canvasFiles);
+    }
+  }, [manifest]);
+
+
+  const handleDownload = (event, file) => {
+    event.preventDefault();
+    fileDownload(file.id, file.filename);
+  };
+
+  return (
+    <div data-testid="supplemental-files" className="ramp--supplemental-files">
+      {showHeading && (
+        <div className="ramp--supplemental-files-heading" data-testid="supplemental-files-heading">
+          <h4>Files</h4>
+        </div>
+      )}
+      <div className="ramp--supplemental-files-display-content" data-testid="supplemental-files-display-content">
+        {Array.isArray(manifestSupplementalFiles) && manifestSupplementalFiles.length > 0 && ( <h4>{itemHeading}</h4> )}
+        {Array.isArray(manifestSupplementalFiles) && manifestSupplementalFiles.length > 0 && (
```

Comment:
Here, instead of doing the check twice is it possible to do it like this?
```
{Array.isArray(manifestSupplementalFiles) && manifestSupplementalFiles.length > 0 && (
  <React.Fragment>
    <h4>{itemHeading}</h4>
    <dl key={'item-files'}>
      { manifestSupplementalFiles.map((file, index) => {
         return (
           <dd key={index}>
             <a href={file.id} key={index} onClick={e => handleDownload(e, file)}>{file.label}</a>
           </dd>);
        })
     }
    </dl>
  </React.Fragment>
)}
```

Reply:
Thanks for the suggestion!  It worked. :tada: